tests/e2e/lib.sh

```diff
@@ -43,42 +43,48 @@ deploy_stackrox() {
-    wait_for_api
-    setup_client_TLS_certs "${2:-}"
-    record_build_info
+    ci_export CENTRAL_CHART_VERSION_OVERRIDE "$1"
+    ci_export SENSOR_CHART_VERSION_OVERRIDE="$2"
+    ci_export DEPLOY_STACKROX_VIA_OPERATOR "false"
+    ci_export OUTPUT_FORMAT "helm"
+    ci_export DISABLE_RHACS_IMAGE_REPOSITORY_PARAMS "true"
+    unset COLLECTOR_IMAGE_REPO
 
-    # generate init bundle
-    password_file="$ROOT/deploy/$ORCHESTRATOR_FLAVOR/central-deploy/password"
-    if [ ! -f "$password_file" ]; then
-        die "password file $password_file not found after deploying central"
-    fi
-    kubectl -n stackrox exec deploy/central -- roxctl --insecure-skip-tls-verify \
-        --password "$(cat "$password_file")" \
-      central init-bundles generate stackrox-init-bundle --output - 1> stackrox-init-bundle.yaml
+    helm repo add stackrox-oss https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource
+    helm repo update
 
-    deploy_sensor_from_helm_charts "$target_version" ./stackrox-init-bundle.yaml
+    latest_tag=$(make tag)
 
-    echo "Sensor deployed. Waiting for sensor to be up"
-    sensor_wait
+    helm_charts=$(helm search repo stackrox-oss -l)
+    central_regex="stackrox-oss/stackrox-central-services[ \t]*.${CENTRAL_CHART_VERSION_OVERRIDE}[ \t]*.([0-9]+\.[0-9]+\.[0-9]+)"
+    sensor_regex="stackrox-oss/stackrox-secured-cluster-services[ \t]*.${SENSOR_CHART_VERSION_OVERRIDE}[ \t]*.([0-9]+\.[0-9]+\.[0-9]+)"
 
-    # Bounce collectors to avoid restarts on initial module pull
-    kubectl -n stackrox delete pod -l app=collector --grace-period=0
+    if  [[ $helm_charts =~ $central_regex ]]; then
+        ci_export CENTRAL_CHART_DIR_OVERRIDE "stackrox-oss/stackrox-central-services"
+    elif [[ "$latest_tag" != "$CENTRAL_CHART_VERSION_OVERRIDE" ]]; then
+        echo >&2 "stackrox-central-services helm chart for version ${CENTRAL_CHART_VERSION_OVERRIDE} not found in stackrox-oss repo nor is it the latest tag."
+        exit 1
+    fi
 
-    sensor_wait
+    if [[ $helm_charts =~ $sensor_regex ]]; then
+        sensor_chart_dir_override="stackrox-oss/stackrox-secured-cluster-services"
+    elif [[ "$latest_tag" == "$SENSOR_CHART_VERSION_OVERRIDE" ]]; then
+        sensor_chart_dir_override="latest-secured-cluster-services"
+        roxctl helm output secured-cluster-services --image-defaults=development_build --output-dir "${sensor_chart_dir_override}" --remove --debug
+        echo "Downloaded stackrox-secured-cluster-services helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} to ${sensor_chart_dir_override}"
+        unset SENSOR_CHART_VERSION_OVERRIDE
+    else
+        echo >&2 "stackrox-secured-cluster-services helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} not found in stackrox-oss repo nor is it the latest tag."
+        exit 1
+    fi
 
-    wait_for_collectors_to_be_operational
+    deploy_stackrox "" "${SENSOR_CHART_VERSION_OVERRIDE:-}" "${sensor_chart_dir_override:-}"
```

Comment:
2. I changed `SENSOR_CHART_VERSION_OVERRIDE` to a positional argument after our discussion, but there were other areas of the code that needed the export so I exported it again. I can see why that is a mismatch so I'll stick with just the export now since it's easier. Specifically the `compatibility_test()` function in `run-compatibility.sh` seemed to require this env variable and I'm not sure how to get it to work without it.
3. Central deployment is a lot more convoluted than central and passing a variable through all of that would make it harder to read and not be worth it in my opinion. For sensor it works, but since we want consistency I'll use an export for sensor again.

I'll bookmark that stackoverflow in my TOREAD folder and check it when I have the time 👍